### ISSUE TITLE
Registrar lock for .se not updating in WHMCS

### ISSUE THREAD

**Issue by DennisSkov** [hexonet/whmcs-ispapi-registrar#164]
When a client or admin enables registrar lock for .se domains, the page refreshes and doesn't show the updated value.
According to the [wiki](https://wiki.hexonet.net/wiki/SE), the .se TLD doesn't support Transfer Lock.
If that's the case, maybe the option should be disabled for .se domains in WHMCS to avoid confusion?

When trying to disable the registrar lock (it's enabled somehow) from the Hexonet control panel, I get this error:
```
[COMMAND]
COMMAND = MODIFYDOMAIN
DELSTATUS0 = clientTransferProhibited
DOMAIN = domain.se
EOF

[RESPONSE]
CODE = 541
DESCRIPTION = Parameter value range error; Currently only clientHold is supported
EOF
```

**Comment by papakai** [comment]
Hi @DennisSkov 

thanks for bringing this up. I've checked the WHMCS Documentation related to that. WHMCS Core just allows to return a domain registrar lock as `unlocked` and `locked`, but not as `unsupported`.
Still, I'll check our possibilities here. It should at least be possible to remove the menu entry `Registrar Lock` on demand.
I guess, we can't influence client area messaging / warnings about the domain being unlocked.

So, basically a design issue of WHMCS Core. I'll keep you posted.

Kai

**Comment by DennisSkov** [comment]
Hi Kai.
What we did to work around this for the DK TLD is using this hook:
```
add_hook('ClientAreaDomainDetailsOutput', 1, function ($domain) {
	// Disable 'Auto Renew' feature for registrar module. If the user accesses #tabAutorenew by link, redirect to the overview
	if($domain['domain']['registrar'] === "registrar") { // Perhaps check if TLD is .se instead
		$html = "<script>if (window.location.href.indexOf('#tabAutorenew') > -1) {
			window.location.replace('#tabOverview');
		}
		$(function() {
			$('#Primary_Sidebar-Domain_Details_Management-Auto_Renew_Settings').addClass('disabled');
		  });</script>";

		return $html;
	}

});
```

It disables the "Auto Renew" menu link, and if the client accesses it using a direct link, they're redirected.

**Comment by papakai** [comment]
Yes, exactly.

A bit off-topic, but of interest for me. May I ask why you did that step with .DK Autorenewal? I know DK Hostmaster is very special and .dk is so often ending in headaches on my end :-) BTW, for 2021 there are plans in queue to review the .DK integration on our side. So probably things will improve hopefully.

**Comment by papakai** [comment]
Looks like this got already covered in the past:
https://github.com/hexonet/whmcs-ispapi-registrar/blob/master/modules/registrars/ispapi/hooks.php#L121

Let me check why it is not working correctly.

**Comment by papakai** [comment]
@DennisSkov enjoy!

**Comment by DennisSkov** [comment]
> A bit off-topic, but of interest for me. May I ask why you did that step with .DK Autorenewal? I know DK Hostmaster is very special and .dk is so often ending in headaches on my end :-) BTW, for 2021 there are plans in queue to review the .DK integration on our side. So probably things will improve hopefully.

We're using our own module for DK Hostmaster and by default, we're set as the Billing Contact.
If the client disables AutoRenew for their .dk domain, WHMCS will no longer generate new invoices but we're still set as the Billing Contact - and in that case they client pays us nothing while we pay DK Hostmaster fees for renewal :-)
Preferably, I would have wanted to be able to hook onto an event when the client disables AutoRenew, and then request DKHM to drop us as the billing contact, but I couldn't find a way to do that. Perhaps you have some ideas? ;-)

Thanks for the update! Will install it ASAP.

**Comment by papakai** [comment]
You could hook into the daily cron, and selecting cases from DB and send an admin mail.
No idea where you can find this in db, I read about WHMCS adds an ActivityLog entry when the autorenewal changes.

http://docs.whmcs.com/Hooks:System#DailyCronJob
